File: backend/src/main/java/com/team05/petmeeting/domain/animal/config/AnimalApiProperties.java
package com.team05.petmeeting.domain.animal.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Getter
@Setter
@ConfigurationProperties(prefix = "animal.api") //application-dev.yml에서 animal.api로 시작하는 프로퍼티를 매핑
public class AnimalApiProperties {
    private String baseUrl; // API의 기본 URL
    private String serviceKey; // API 인증에 필요한 서비스 키
    private String returnType; // API 응답 형식 (예: JSON, XML)
    private int timeoutMs = 5000;
}

File: backend/src/main/java/com/team05/petmeeting/domain/animal/config/AnimalApiProperties.kt
package com.team05.petmeeting.domain.animal.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "animal.api")
class AnimalApiProperties {
    // 외부 유기동물 API 기본 URL
    var baseUrl: String = ""
    // 외부 API 인증 서비스 키
    var serviceKey: String = ""
    // 외부 API 응답 형식
    var returnType: String = ""
    // 외부 API 연결/응답 타임아웃
    var timeoutMs: Int = 5000
}
